psutil/_psutil_osx.c

```diff
@@ -860,6 +874,50 @@ psutil_boot_time(PyObject *self, PyObject *args) {
     return Py_BuildValue("f", (float)boot_time);
 }
 
+/*
+ * Return a Python float indicating the value of the temperature
+ * measured by an SMC key
+ */
+static PyObject *
+psutil_smc_get_temperature(PyObject *self, PyObject *args) {
+    char* key;
+    float temp;
+
+    if (! PyArg_ParseTuple(args, "s", &key)) {
+        return NULL;
+    }
+    temp = SMCGetTemperature(key);
+    return Py_BuildValue("d", temp);
+}
+
+/*
+ * Return a Python float indicating the number of fans in the system
+ */
+static PyObject *
+psutil_fans_count(PyObject *self, PyObject *args) {
+    int fan_count;
+
+    fan_count = SMCGetFanNumber(SMC_KEY_FAN_NUM);
+    if (fan_count < 0) {
+        fan_count = 0;
+    }
+    return Py_BuildValue("i", fan_count);
+}
+
+/*
+ * Return a Python float indicating the fan speed of the number of fan passed
+ */
+static PyObject *
+psutil_get_fan_speed(PyObject *self, PyObject *args) {
+    int key;
+    float speed;
+
+    if (! PyArg_ParseTuple(args, "i", &key)) {
+        return NULL;
+    }
+    speed = SMCGetFanSpeed(key);
+    return Py_BuildValue("f", speed);
+}
```

Comment:
I suggest to use a single C `psutil_sendors_fans()` function which gets the number of fans and their speed in one shot. I see no need of 2 separate functions.